src/tracer.js

```diff
@@ -228,6 +228,15 @@ export default class Tracer {
             ctx.parentId = null;
             ctx.flags = flags;
         } else {
+            if (!parent.samplingFinalized) {
+                if (this._sampler.isSampled(operationName, internalTags)) {
+                   parent._flags |= constants.SAMPLED_MASK
+                } else {
+                    parent._flags &= ~constants.SAMPLED_MASK
+                }
+                parent.finalizeSampling()
```

Comment:
We could have another boolean in the parent span to store the state of whether all child spans must be sampled and update that instead of the parent span context. Would you prefer the implementation to do that?